### PR TITLE
[ MB-15049 ] Create weights moved table component

### PR DESCRIPTION
**Pull request by mmayo-truss**
## Summary

[Here is a link to the relevant Figma](https://www.figma.com/file/5AhuEkaH3bJVrtffU3wp22/Milmove-Office-App?node-id=8396%3A101275&t=LJTVn9BflXJqw2Dj-4)

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

</details>

### Additional steps

1. Go to Review Shipment Weights Table component in storybook

## Screenshots
PPM weights moved table: 
image

Pro-gear table: 
image

Non-PPM weights moved table: 
image

### CODE REVIEW

File: src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useFilters, usePagination, useSortBy, useTable } from 'react-table';
+
+import styles from './ReviewShipmentWeightsTable.module.scss';
+import { addShipmentNumbersToTableData } from './helpers';
+
+import Table from 'components/Table/Table';
+
+const ReviewShipmentWeightsTable = (props) => {
```
Comment (michaelvaldes-truss) [inline src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx:9]: **suggestion:** should we break down the props explicitly into tableData and tableConfig directly in the arguments?

File: src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
```
Comment (michaelvaldes-truss) [inline src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx:1]: **suggestion (non-blocking):** does it make sense to put this entire component directory in the src/pages/Office/ServicesCounselingReviewShipmentWeights directory (the directory of its parent page)? The src/pages/Office directory structure is pretty flat right now, but I don't know if that's necessary/intended.

**Review by michaelvaldes-truss — COMMENTED** [review]
Looking great so far! The separation of helpers is super useful.

File: src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx (hunk 1 above)
Comment (mmayo-truss) [reply src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx]: Good Idea. Done-zo!

File: src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx (hunk 2 above)
Comment (mmayo-truss) [reply src/components/Office/PPM/ReviewShipmentWeightsTable/ReviewShipmentWeightsTable.jsx:1]: I could go either way, though after a cursory look, I am not seeing any storybook files in any other directory other than `components`.

File: src/components/Office/PPM/ReviewShipmentWeightsTable/helpers.jsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    // We only show the number for the shipment
+    // if there are more than one of that type in the table data.
+    const showNumber = shipmentCountByType[shipmentType] > 1;
+
+    return { ...shipment, showNumber, shipmentNumber };
+  });
+  return modifiedTableData;
+}
+
+export function determineTableRowClassname(shipmentType) {
```
[the remark this answers is not in the document]
Comment (mmayo-truss) [reply src/components/Office/PPM/ReviewShipmentWeightsTable/helpers.jsx:54]: That is a good point. I did have that at first, however I switched it to [mimic the way ShipmentListItem handles it](https://github.com/transcom/mymove/blob/831b433b6faf80dd2a50dba24911f87ebbcf5c54/src/components/ShipmentList/ShipmentList.jsx#L28-L36).  I can switch that back.

[the remark this answers is not in the document]
Comment (afoon) [reply src/components/Office/PPM/ReviewShipmentWeightsTable/helpers.jsx:53]: **praise**: While this more verbose, it is easier for a junior engineer to understand what this util is supposed to do 🎉

**Review by afoon — APPROVED** [review]
LGTM. I left some clarification on my last change suggestion but it doesn't affect the overall component shipping 🚢

File: src/components/Office/PPM/ReviewShipmentWeightsTable/helpers.jsx [hunk 4]
```diff
@@ -107,7 +117,7 @@ export function actualWeightDisplayHelper(row) {
   return actualWeight > 0 ? formatWeight(actualWeight) : DASH;
 }
 
-export const NoRowsMessages = {
+export const noRowsMessages = {
```
[the remark this answers is not in the document]
Comment (mmayo-truss) [reply src/components/Office/PPM/ReviewShipmentWeightsTable/helpers.jsx]: Thanks for that clarification @amina. I have addressed this.

**Review by AnnaGingle — APPROVED** [review]
This is beautiful. Great work!!! APPROVED.

**Pull request merged**